CONTRIBUTING.md

```diff
@@ -1,95 +1,149 @@
+Issues will be marked with different labels for different purpose, see [Labels](#labels) for details.
 
-## Contributing a patch for a known bug, or a small correction
+### Issue Lifecycle
 
-You should follow the basic GitHub workflow:
+The issue lifecycle is mainly driven by the maintainers.
 
- 1. fork
- 2. commit a change
- 3. make sure the tests pass
- 4. PR
+1. Issue creation:
+   - After an issue is created, it will be labelled `bug` or `enhancement`, and `triage`.
+2. Triage: The maintainers will triage the issues. The results of triage could be one of the following:
+   - Remove label `triage`, and add the issues to the correct milestone
+   - Keep label `triage` to indicate that further triage is required.
+   - Remove label `triage`, and labelled `wont fix` to indicate this issue is out of the scope of Notary or other reasons.
+3. Development:
+   - An issue labelled `enhancement` normally require solution proposal and spec updates if needed before implementation
+   - An issue that is labeled as `enhancement` or `bug` should be connected to the PR that resolves it
+4. Issue closure
+   - Close the issue once PRs are merged
+   - Close the issue once the author accepts the `wont fix` decision.
 
-Additionally, you must [sign your commits](https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work). It's very simple:
+## Support requests or questions
 
- - configure your name with git: `git config user.name "Real Name" && git config user.email mail@example.com`
- - sign your commits using `-s`: `git commit -s -m "My commit"`
+Please use slack channel for support requests or questions. Issues could be created accordingly if it is a bug or enhancement.
 
-Some simple rules to ensure quick merge:
+## Contributing Workflow
 
- - clearly point to the issue(s) you want to fix in your PR comment (e.g., `closes #12345`)
- - prefer multiple (smaller) PRs addressing individual issues over a big one trying to address multiple issues at once
- - if you need to amend your PR following comments, please squash instead of adding more commits
- - if fixing a bug or adding a feature, please add or update the relevant `CHANGELOG.md` entry with your pull request number
-   and a description of the change
+PR are always welcome, even if they only contain small fixes like typos or a few lines of code. If there will be a significant effort, please document it as an issue and get a discussion going before starting to work on it.
+
+Please submit a PR broken down into small changes bit by bit. A PR consisting of a lot of features and code changes may be hard to review. It is recommended to submit PRs in an incremental fashion.
+
+Note: If you split your pull request to small changes, please make sure any of the changes goes to main will not break anything. Otherwise, it can not be merged until this feature complete.
+
+- Fork and clone
```

Comment:
Consider `###` instead of `-`.

---

For example:

### Fork and clone
Fork the repository and clone the code to your local workspace.

### Branch
Changes should be made on your own fork in a new branch checked out from the `main` branch.